File: Problem002.java
public class Problem002 {
    public static void main(String[] args) {
        int n1;
        int n2 = 1;
        int n3 = 2;
        int sum = 0;

        while (n3 <= 4000000) {
            sum = sum + n3;
            n1 = n2 + n3;
            n2 = n3 + n1;
            n3 = n1 + n2;
        }

        System.out.println(sum);
    }
}

File: go/Problem002.go
package main

import "fmt"

func main() {
	a1 := 1
	a2 := 2
	var a3 int
	sum := 0

	for a2 <= 4000000 {
		sum += a2
		a3 = a1+a2
		a1 = a2+a3
		a2 = a3+a1
	}

	fmt.Println(sum)
}
